### PR TITLE
MM-36259 - Navigation between playbook backstage screens

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- playbook edit: new 'cancel' button
- playbook edit: clicking 'save' or 'cancel' will return to playbook landing page
- playbook edit: remove left-arrow, increase max-width for title in edit screen (we have extra space without the left-arrow)
- playbook list screen: edit action should go to playbook edit
- incident overview: left arrow will bring you to ~~playbook landing page~~ incident list

Where we are:
- [x] Backend stats, rough frontend. 
- [x] New navigation model (between list, landing, edit, messaging) <-- We are here
- [ ] UI tweaks to bring into 100% in line with designs
- [ ] license checked and upgrade bannered

image

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-36259

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/playbook_edit.tsx [hunk 1]
```diff
@@ -621,7 +629,7 @@ const PlaybookEdit: FC<Props> = (props: Props) => {
                 title={'Confirm discard'}
                 message={'Are you sure you want to discard your changes?'}
                 confirmButtonText={'Discard Changes'}
-                onConfirm={props.onClose}
+                onConfirm={() => onClose()}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/playbook_edit.tsx:632]: One of my classic nits (feel free to ignore, this is just me being super picky):
```suggestion
                onConfirm={onClose}
```

**Review by agarciamontoro — APPROVED** [review]
Looks great! 

One consideration: when clicking on the X to close the backstage, the playbook edit screen does not warn about losing the changes made. I think this has been always the case, but just in case.

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
The described functionality looks good @cpoile! 

Can we style the `Cancel` button as a secondary button and reverse the order? Thanks!
image

**Comment by cpoile** [comment]
> when clicking on the X to close the backstage, the playbook edit screen does not warn about losing the changes made. I think this has been always the case, but just in case.

Good catch, I opened a tech debt ticket for that: https://mattermost.atlassian.net/browse/MM-36322

File: webapp/src/components/backstage/playbook_edit.tsx (hunk 1 above)
Comment (cpoile) [reply webapp/src/components/backstage/playbook_edit.tsx:632]: Almost, but that won't work here; `onConfirm` tries to pass in a `checked: boolean` which conflicts with the onClose signature. :)

**Comment by cpoile** [comment]
@abhijit-singh Sure! Here you go:
image

**Review by abhijit-singh — APPROVED** [review]
Awesome, thanks!

File: webapp/src/components/backstage/playbook_edit.tsx (hunk 1 above)
Comment (agarciamontoro) [reply webapp/src/components/backstage/playbook_edit.tsx:641]: Oh no :( I don't understand why it lets you pass a function with no arguments, then.  the type safety

**Comment by abhijit-singh** [comment]
> when clicking on the X to close the backstage, the playbook edit screen does not warn about losing the changes made. I think this has been always the case, but just in case.

@agarciamontoro I do see that warning currently

**Comment by cpoile** [comment]
@abhijit-singh Found the problem: master changed. I need to rebase and fix some merge conflicts. This will be solved.

**Review by iomodo — APPROVED** [review]
LGTM! Thanks @cpoile

**Pull request merged**